### PR TITLE
feat: add `--add-ingress-from` flag to `storage init`

### PR DESCRIPTION
**Pull request by Lou1415926**
The flag will produce the `AccessPolicy` or  `SecurityGroupIngress` as a workload-level addon for the workload to access an env-level storage.

This PR includes the `Validate` and `Ask` part of the flag.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/cli.go [hunk 1]
```diff
@@ -129,12 +128,14 @@ func indentListItem(multiline string) string {
 	return strings.Join(prefixedLines, "\n")
 }
 
-func quoteStringSlice(in []string) []string {
-	quoted := make([]string, len(in))
+type stringMutator func(string) string
+
+func mutateStringSlice(in []string, mutate stringMutator) []string {
```
Comment (efekarakus) [inline internal/pkg/cli/cli.go]: I feel like maybe this is one of the occasions where generics are OK, what do you think of:
```go
func applyAll[T any](in []T, fn func(item T) T) []T {
	out := make([]T, len(in))
	for i, v := range in {
		out[i] = fn(v)
	}
	return out
}

File: internal/pkg/cli/flag.go [hunk 2]
```diff
@@ -284,9 +286,11 @@ Uploaded asset locations are filled in the template configuration.`
 	pipelineTypeFlagDescription      = `The type of pipeline. Must be either "Workloads" or "Environments".`
 
 	// Storage.
-	storageFlagDescription             = "Name of the storage resource to create."
-	storageWorkloadFlagDescription     = "Name of the service or job to associate with storage."
-	storageLifecycleFlagDescription    = "Whether the storage should be created and deleted at the same time as a workload or as the environment"
+	storageFlagDescription               = "Name of the storage resource to create."
+	storageWorkloadFlagDescription       = "Name of the service or job to associate with storage."
+	storageLifecycleFlagDescription      = "Whether the storage should be created and deleted at the same time as a workload or as the environment"
+	storageAddIngressFromFlagDescription = `The workload that needs access to an environment-level storage.
```
Comment (efekarakus) [inline internal/pkg/cli/flag.go]:
```suggestion
	storageAddIngressFromFlagDescription = `The workload that needs access to an environment-level addon.
```

File: internal/pkg/cli/storage_init.go [hunk 3]
```diff
@@ -139,11 +140,19 @@ var engineTypes = []string{
 	engineTypePostgreSQL,
 }
 
+var (
+	ddbFlags                       = []string{storagePartitionKeyFlag, storageSortKeyFlag, storageNoSortFlag, storageLSIConfigFlag, storageNoLSIFlag}
+	ddbFlagExclusiveWithAddIngress = ddbFlags
```
Comment (efekarakus) [inline internal/pkg/cli/storage_init.go]: Can we use cobra's [MarkFlagsMutuallyExclusive](https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsMutuallyExclusive) method and get rid of the internal bool flag `configFlagExclusiveWithAddIngress`?

Comment (Lou1415926) [reply internal/pkg/cli/storage_init.go]: done!

**Review by efekarakus — APPROVED** [review]

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**